.github/workflows/main.yml

```diff
@@ -127,15 +119,16 @@ jobs:
           - stable-2.9
           - stable-2.10
           - stable-2.11
+          - stable-2.12
           - devel
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python
-        uses: actions/setup-python@v2.2.2
+        uses: actions/setup-python@v2
         with:
-          python-version: "3.7"
+          python-version: "3.8"
       - name: Install Ansible
-        run: pip install --upgrade git+https://github.com/ansible/ansible.git@${{ matrix.ansible }}
+        run: pip install --upgrade -r ansible
```

Comment:
this looks wrong. `-r` means requirements file, but ansible is none.

and without `-r` this would install *some* ansible, not the one we want in this matrix